examples/example-tls/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientS2A.java

```diff
@@ -0,0 +1,118 @@
+import javax.net.ssl.SSLException;
+import javax.annotation.Nonnull;
+
+// TODO: change flags to use com.google.common.flags
+
+/**
+ * Establishes a TLS 1.3 connection with a greeter service.
+ *
+ * args:
+ * [ServerAddress] In the form of host:port
+ * [clientCertChainFilePath] File Path to Client Certificate
+ * [clientPrivateKeyFilePath] File Path to Client Key
+ * [trustCertCollectionFilePath] File Path to CA Certificate
+ */
+public class HelloWorldClientS2A {
+    private static final Logger logger = Logger.getLogger(HelloWorldClientS2A.class.getName());
+    private final ManagedChannel channel;
+    private GreeterGrpc.GreeterBlockingStub blockingStub;
+
+    private static SslContext buildSslContext(String clientCertChainFilePath,
+                                              String clientPrivateKeyFilePath,
+                                              String trustCertCollectionFilePath) throws SSLException {
+      checkNotNull(clientCertChainFilePath, "clientCertChainFilePath");
+      checkNotNull(clientPrivateKeyFilePath, "clientPrivateKeyFilePath");
+      checkNotNull(trustCertCollectionFilePath, " trustCertCollectionFilePath");
+      return GrpcSslContexts.forClient()
+            .trustManager(new File(trustCertCollectionFilePath))
+            .keyManager(new File(clientCertChainFilePath), new File(clientPrivateKeyFilePath))
+            .protocols("TLSv1.3")
+            .build();
+    }
+
+    /**
+     * Constructs client for accessing greeter server using the existing channel.
+     */
+    private HelloWorldClientS2A(ManagedChannel channel, GreeterGrpc.GreeterBlockingStub blockingStub) {
+        this.channel = channel;
+        this.blockingStub = blockingStub;
+    }
+
+    private void shutdown() throws InterruptedException {
+        channel.shutdown().awaitTermination(5, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Say hello to server.
+     */
+    private void greet(String name) {
+        logger.info("Will try to greet " + name + " ...");
+        HelloRequest request = HelloRequest.newBuilder().setName(name).build();
+        HelloReply response;
+        try {
+            response = blockingStub.sayHello(request);
+        } catch (StatusRuntimeException e) {
+          logger.log(Level.WARNING, "RPC failed: {0}", e.getStatus());
+            return;
+        }
+        logger.info("Greeting: " + response.getMessage());
+    }
+
+    private static HelloWorldClientS2A create(String host, int port, SslContext sslContext) throws SSLException{
+        checkNotNull(host, "host");
+        checkNotNull(port,"port");
+        ManagedChannel channel = NettyChannelBuilder.forAddress(host,port)
+                        .sslContext(sslContext)
+                        .build();
+        return new HelloWorldClientS2A(channel, GreeterGrpc.newBlockingStub(channel));
+    }
+
+    /**
+     * Greet server. If provided, the first element of {@code args} is the name to use in the
+     * greeting.
```

Comment:
If `args[0]` is not set, will this code crash? If so, can we provide a default value to avoid that?

Reply:
Please respond to this comment.

Reply:
Original sentence was left over from the very original greeter file; the default args have the host and port be separate arguments. 

I removed it since it was inconsistent with our intended args--we pass in address, which is "host:port" as the first element, and is treated as the 'name'

Reply:
This comment is superseded by a later one, where I ask about adding default values for all of the parameters. Let's resolve this comment and can use the other one for tracking instead.